### PR TITLE
feat: add docker compatibility button in settings

### LINKED ISSUE #3709: Compose onboarding: Show the compatibility mode
### Is your enhancement related to a problem? Please describe

As per https://github.com/containers/podman-desktop/issues/2523:

I want to know HOW to use Compose and if compatibility mode is enabled or disabled and what does that mean.

### Describe the solution you'd like

Will be added as part of the "onboarding" steps.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #3851: Compose: Add "docker compatibility enable" button as a configurable button in preferences.
### Is your enhancement related to a problem? Please describe

Similar to https://github.com/containers/podman-desktop/issues/3810 we should be able to add "docker compatibility enable" as a toggle *button* within preferences. 

See below similar to "system wide install"
image

### Describe the solution you'd like

Compatibility enablement moved into preferences rather than through the current status-bar-like enablement that we currently have.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cdrage**
feat: add docker compatibility button in settings

### What does this PR do?

This PR adds a "docker compatibility" enable and disable button within
the Podman extension settings rather than having to go through the
status bar in order to enable / disable the setting.

### Screenshot/screencast of this PR

https://github.com/containers/podman-desktop/assets/6422176/902b2566-62ad-4ee6-87af-6a9a2705de99

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop/issues/3851
Closes https://github.com/containers/podman-desktop/issues/3709

### How to test this PR?

Go into settings and click enable / disable on the button.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
I don't believe tests are required since this is simply calling the same functions that are already covered within `compatibility-mode.spec.ts`.

Let me know if this appropriate to have in the settings section @benoitf @deboer-tim since this is a bit different than the system-wide-install button.

**Comment by benoitf** [comment]
hello, IMHO it requires unit tests to ensure that it performs the right logic when configuration is updated, etc.

new methods with logic (not only pass-throughs) == new tests required

**Comment by cdrage** [comment]
> hello, IMHO it requires unit tests to ensure that it performs the right logic when configuration is updated, etc.
> 
> new methods with logic (not only pass-throughs) == new tests required

No problem! I'll test to make sure `activate()` calls the handler calls, etc.

Does the implementation look good / having the compatibility button within the settings makes sense?

**Comment by cdrage** [comment]
Took a while but tests have been added and ready for another review @benoitf @lstocchi

File: extensions/podman/package.json
Comment (lstocchi) [inline extensions/podman/package.json]: This is not supported i guess. I can see the setting on Windows 😆

Comment (cdrage) [reply extensions/podman/package.json]: Okay I'll open up an issue and work on it now.

**Comment by cdrage** [comment]
https://github.com/containers/podman-desktop/pull/4159 opened, this PR is blocked until https://github.com/containers/podman-desktop/pull/4159 is in and then this feature will work and show on Linux and Mac and not Windows.

**Comment by cdrage** [comment]
Unblocked ready for review 👍

**Comment by lstocchi** [comment]
@cdrage can you please rebase? 🙏

**Comment by cdrage** [comment]
> @cdrage can you please rebase? 🙏

Done 👍

File: extensions/podman/src/extension.ts
Comment (lstocchi) [inline extensions/podman/src/extension.ts]: Should we move it in a place that it's only executed in a non-windows system?

**Review by lstocchi — APPROVED** [review]
LGTM

**Comment by cdrage** [comment]
@lstocchi You're right, it should only do the `extensionApi.configuration.onDidChangeConfiguration` if it's on Linux or Mac, I've moved that part of the code under `if (isMac() || isLinux()) {`. 👍

**Review by dgolovin — APPROVED** [review]
Works for me.

**Pull request merged**
